content/docs/specs/remote_write_spec_2_0.md

```diff
@@ -130,18 +130,43 @@ Senders MUST include a user agent header that SHOULD follow [the RFC 9110 User-A
 
-The following subsections specify Sender and Receiver semantics around different write error cases.
+The following subsections specify Sender and Receiver semantics around headers and different write error cases.
+
+#### Required `X-Prometheus-Remote-Write-Written-*` Response Headers
+
+<!---
+Rationales: https://github.com/prometheus/prometheus/issues/14359
+-->
+Upon a successful content negotiation, Receivers process (write) the received batch of data. Once completed (with success or failure) for each important piece of data (currently Samples, Histograms and Exemplars) Receivers MUST send a dedicated HTTP `X-Prometheus-Remote-Write-Written-*` response header with the precise number of successfully written elements. 
+
+Each header value MUST be a single 64-byte integer. The header names MUST be as follows:
+
+* `X-Prometheus-Remote-Write-Written-Samples <integer; count of all successfully written Samples from this request>`
+* `X-Prometheus-Remote-Write-Written-Histograms <integer; count of all successfully Histogram samples from this request>`
+* `X-Prometheus-Remote-Write-Written-Exemplars <integer; count of all successfully Exemplars from this request>`
+
+Senders MAY use those headers to confirm which parts of data were successfully written by the Receiver. Common use cases:
+
+* [Partial Write](#partial-write) failure situations: Senders MAY use those headers for more accurate client instrumentation and error handling.
+* Broken 1.0 Receiver implementations: Senders SHOULD assume [415 HTTP Unsupported Media Type](https://www.rfc-editor.org/rfc/rfc9110.html#name-415-unsupported-media-type) status code when sending the data using `io.prometheus.write.v2.Request` request and receiving 2xx HTTP status code, but none of the `X-Prometheus-Remote-Write-Written-*` response headers from the Receiver. This is a common issue for the 1.0 Receivers that do not check the `Content-Type` request header; accidental decoding of the `io.prometheus.write.v2.Request` payload with `prometheus.WriteRequest` schema results in empty result and no decoding errors.
+* Unknown broken implementations or issues: Senders MAY use those headers to detect broken Sender and Receiver implementations or other problems.
+
+On the 2xx HTTP status code, Senders CAN assume that a missing write response header means that no such a piece of data was written by the Receiver (count of `0`). value. One exception is that Senders MUST NOT assume the same (empty header means zero) when using the deprecated `prometheus.WriteRequest` Protobuf Message in the request.
+
+Senders MUST NOT assume what Remote Write specification version the Receiver implements from the remote write response headers.
```

Comment:
```suggestion
Senders MUST NOT assume what Remote Write specification version the Receiver implements the remote write response headers.
```

Reply:
actually that's wrong suggestion no? (: I merged this but will revert, it should be "from"?